base/vault-pki.yaml

```diff
@@ -0,0 +1,148 @@
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  name: vault-pki
+---
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: vault-pki
+spec:
+  selector:
+    matchLabels:
+      app: vault-pki
+  template:
+    metadata:
+      labels:
+        app: vault-pki
+    spec:
+      serviceAccountName: vault-pki
+      containers:
+        - name: pki-manager
+          image: everpeace/curl-jq
+          command:
+            - "sh"
+            - "-c"
+            - >
+              echo "${GENERATE_CERTS}" > generate_certs;
+              chmod +x ./generate_certs;
+              ./generate_certs;
+          env:
+            - name: GENERATE_CERTS
+              value: |
+                #!/usr/bin/env bash
+
+                # Validation and defaults
+                : ${VAULT_NAMESPACE:?"Need to set VAULT_NAMESPACE"}
+                : ${VAULT_CLIENT_NAMESPACES:?"Need to set VAULT_CLIENT_NAMESPACES"}
+                kubectl_version="${KUBECTL_VERSION:-"$(curl -s https://storage.googleapis.com/kubernetes-release/release/stable.txt)"}"
+                secret_name="${VAULT_SECRET_NAME:-"vault-tls"}"
+                vault_name="${VAULT_NAME:-"vault"}"
+                replicas="${VAULT_REPLICAS:-"3"}"
+
+                if ! which curl > /dev/null; then
```

Comment:
`curl` is used in setting `kubectl_version`, I think this check should be first?

Reply:
good point